### PR TITLE
add dns resolve time , add hostname to label, reduce MinIntervalSeconds

### PR DESCRIPTION
**Pull request by yonghua-sun**

### CODE REVIEW

File: plugins/input/netping/netping.go [hunk 1]
```diff
@@ -115,38 +129,52 @@ func (m *NetPing) Init(context ilogtail.Context) (int, error) {
...
 	localICMPConfigs := make([]ICMPConfig, 0)
 
+	m.resolveHostMap = &sync.Map{}
+
 	for _, c := range m.ICMPConfigs {
-		if c.Src == localIP {
+		if c.Src == m.ip {
 			localICMPConfigs = append(localICMPConfigs, c)
 			m.hasConfig = true
+			m.resolveHostMap.Store(c.Target, "")
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:148]: should remove this line?

File: plugins/input/netping/netping.go [hunk 2]
```diff
@@ -115,38 +129,52 @@ func (m *NetPing) Init(context ilogtail.Context) (int, error) {
...
 	m.processTimeoutAndInterval()
 
-	localIP := util.GetIPAddress()
+	m.ip = util.GetIPAddress()
 
 	m.hasConfig = false
 	// get local icmp target
 	localICMPConfigs := make([]ICMPConfig, 0)
 
+	m.resolveHostMap = &sync.Map{}
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:142]: why need sync map ?

File: docs/en/plugins/metric_input/metric_input_netping.md [hunk 3]
```diff
@@ -4,7 +4,8 @@ a icmp-ping/tcp-ping plugin for logtail
 ## Config
 |  field   |   type   |   description   | default value   |
 | ---- | ---- | ---- | ---- |
+|disable_dns_metric|bool|disable dns resolve metric, default is false| false|
```
Comment (EvanLjp) [inline docs/en/plugins/metric_input/metric_input_netping.md:7]: Please add change log

Comment (EvanLjp) [inline docs/en/plugins/metric_input/metric_input_netping.md:7]: Please add change log

File: plugins/input/netping/netping.go [hunk 4]
```diff
@@ -155,12 +183,79 @@ func (m *NetPing) Description() string {
...
+		if err == nil {
+			n = nBig.Int64()
+		}
+		m.resolveHostMap.Store(host, ips[n].String())
+	}
+
+	var label helper.KeyValues
+	label.Append("target", host)
+	label.Append("src", m.hostname)
+	label.Append("src_host", m.hostname)
```
Comment (messixukejia) [inline plugins/input/netping/netping.go:206]: src and src_host both are m.hostname?

File: plugins/input/netping/netping_test.go [hunk 5]
```diff
@@ -286,3 +321,19 @@ func TestProcessTimeoutAndInterval(t *testing.T) {
...
+
+	ctx := mock.NewEmptyContext("project", "store", "config")
+	netPing := &NetPing{}
+
+	netPing.Init(ctx)
+
+	netPing.evaluteDNSResolve("www.baidu.com")
+
+	result := <-netPing.resolveChannel
+	fmt.Println(result)
```
Comment (messixukejia) [inline plugins/input/netping/netping_test.go:336]: Should the result be serialized?

File: plugins/input/netping/netping.go [hunk 6]
```diff
@@ -115,38 +129,52 @@ func (m *NetPing) Init(context ilogtail.Context) (int, error) {
...
+
 	for _, c := range m.ICMPConfigs {
-		if c.Src == localIP {
+		if c.Src == m.ip {
 			localICMPConfigs = append(localICMPConfigs, c)
 			m.hasConfig = true
+			m.resolveHostMap.Store(c.Target, "")
+
+			if !m.DisableDNS {
+				m.resolveHostMap.Store(c.Target, "")
```
Comment (messixukejia) [inline plugins/input/netping/netping.go:151]: store twice?

File: plugins/input/netping/netping.go [hunk 7]
```diff
@@ -155,12 +183,79 @@ func (m *NetPing) Description() string {
...
+		Success: success,
+		RTMs:    float64(rt.Milliseconds()),
+		Label:   label.String(),
+	}
+}
+
+func (m *NetPing) getRealTarget(target string) string {
+	realTarget := ""
+	v, exists := m.resolveHostMap.Load(target)
+	if exists {
```
Comment (messixukejia) [inline plugins/input/netping/netping.go:218]: if else？

File: plugins/input/netping/netping_test.go (hunk 5 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping_test.go]: modified

File: plugins/input/netping/netping.go (hunk 4 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:197]: changed

File: docs/en/plugins/metric_input/metric_input_netping.md (hunk 3 above)
Comment (yonghua-sun) [reply docs/en/plugins/metric_input/metric_input_netping.md:7]: added

File: plugins/input/netping/netping.go (hunk 1 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: removed

File: plugins/input/netping/netping.go (hunk 6 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: removed

File: plugins/input/netping/netping.go (hunk 7 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:209]: else will  be covered

	if realTarget == "" {
		realTarget = target
	}

**Review by EvanLjp — APPROVED** [review]

**Pull request merged**